pulsar-broker/src/test/java/org/apache/pulsar/client/api/TopicReaderTest.java

```diff
@@ -359,4 +362,59 @@ public EncryptionKeyInfo getPrivateKey(String keyName, Map<String, String> keyMe
         reader.close();
         log.info("-- Exiting {} test --", methodName);
     }
+
+
+    @Test
+    public void testSimpleReaderReachEndofTopic() throws Exception {
```

Comment:
One scenario in which we need to pay attention is when batches are used. 

With batches, multiple messages are stored in a single BK entry, and the broker treat a batch as a unit, which the consumer will finally break up, presenting the individual messages to application.

Reply:
thanks, will add the test.